src/linker/Linker/CustomAttributeSource.cs

```diff
@@ -40,8 +40,13 @@ public bool TryGetEmbeddedXmlInfo (ICustomAttributeProvider provider, [NotNullWh
 			var assembly = GetAssemblyFromCustomAttributeProvider (provider);
 
 			if (!_embeddedXmlInfos.TryGetValue (assembly, out xmlInfo)) {
+				// Add an empty record - this prevents reentrancy
+				// If the embedded XML itself generates warnings, trying to log such warning
+				// may ask for attributes (suppressions) and thus we could end up in this very place again
+				// So first add a dummy record and once processed we will replace it with the real data
```

Comment:
Makes sense, but this scares me 😨

Reply:
Yeah - I'm not happy about it either. I did go through the code which can run here, and there's only very few places where we generate warnings, so it's not that bad. Maybe we should think about this some more how to solve - but this is hard, we want to be lazy about the suppressions (too expensive to do eagerly).